api/controller/migrationtarget/client.go

```diff
@@ -106,28 +104,21 @@ func (c *Client) Activate(modelUUID string, sourceInfo coremigration.SourceContr
 
 // UploadCharm sends the content to the API server using an HTTP post in order
 // to add the charm binary to the model specified.
-func (c *Client) UploadCharm(modelUUID string, curl *charm.URL, content io.ReadSeeker) (*charm.URL, error) {
-	args := url.Values{}
-	args.Add("name", curl.Name)
-	args.Add("schema", curl.Schema)
-	args.Add("arch", curl.Architecture)
-	args.Add("series", curl.Series)
-	args.Add("revision", strconv.Itoa(curl.Revision))
-
-	apiURI := url.URL{
-		Path:     "/migrate/charms",
-		RawQuery: args.Encode(),
-	}
```

Comment:
Not at the moment, although that is the long term goal for Juju 4

For the time being, similar to the `/model-:uuid/charms/:object` endpoint, we send the whole Charm URL string in a header `Juju-Curl`, which the apiserver then parses if we need any of these values. 

This is required since, in 3.4, the handler for this endpoint just shims around the old handler, where each element of the url are broken up. This will be properly implemented in 4.0

The hope is by Juju 4.0 we will no longer use the charm url for these values, and will ignore the header.